### PR TITLE
Extend API with contacts, fix refresh token bug, allow filter listst

### PR DESCRIPTION
**Pull request by Imperatus**

### CODE REVIEW

**Comment by wdoekes** [comment]
Thanks for the PR. I'll look into this this week.

File: README.rst [hunk 1]
```diff
@@ -292,6 +292,7 @@ Currently, it looks like this:
     ledgeraccounts = LedgerAccounts.as_property()
     receivables = Receivables.as_property()
     relations = Relations.as_property()
+    contacts = Contacts.as_property()
```
Comment (wdoekes) [inline README.rst]: Please order alphabetically.

File: exactonline/api/__init__.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 from .ledgeraccounts import LedgerAccounts
 from .receivables import Receivables
 from .relations import Relations
+from .contacts import Contacts
```
Comment (wdoekes) [inline exactonline/api/__init__.py]: Please order alphabetically.

File: exactonline/api/__init__.py [hunk 3]
```diff
@@ -35,3 +36,4 @@ class ExactApi(
     ledgeraccounts = LedgerAccounts.as_property()
     receivables = Receivables.as_property()
     relations = Relations.as_property()
+    contacts = Contacts.as_property()
```
Comment (wdoekes) [inline exactonline/api/__init__.py]: Please order alphabetically.

File: exactonline/api/contacts.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# vim: set ts=8 sw=4 sts=4 et ai tw=79:
+"""
+Helper for relation/contact resources.
+
+This file is part of the Exact Online REST API Library in Python
+(EORALP), licensed under the LGPLv3+.
+Copyright (C) 2015 Walter Doekes, OSSO B.V.
```
Comment (wdoekes) [inline exactonline/api/contacts.py]: 2017

File: exactonline/api/manager.py [hunk 5]
```diff
@@ -71,6 +71,9 @@ def filter(self, **kwargs):
         # kwargs = {'filter': "EntryDate+gt+datetime'2014-01-01'", 'top': 5}
         args = []
         for key, value in kwargs.items():
+            if isinstance(value, list):
+                value = ','.join(value)
```
Comment (wdoekes) [inline exactonline/api/manager.py]: This should be a separate PR with accompanying addition to the README.

File: exactonline/rawapi.py [hunk 6]
```diff
@@ -158,6 +158,10 @@ def _set_tokens(self, jsondata):
         #  "token_type":"bearer",
         #  "expires_in":"600",
         #  "refresh_token":"__1P!I.."}
+
+        # Token refresh call somehow returns bytes
+        if isinstance(jsondata, bytes):
+            jsondata = jsondata.decode('utf-8')
```
Comment (wdoekes) [inline exactonline/rawapi.py]: Not needed anymore since e436fd2c4c018d9c21cd18fd8ad7c3a45d471369

**Comment by wdoekes** [comment]
Looks good to me. Please squash into a single commit (git rebase with fixups) with a good commit message.

**Comment by Imperatus** [comment]
Sorry for the git-fu. Should be a single commit now. Please confirm if it is up to your standards.
